public/dorequest.php

```diff
@@ -368,6 +369,12 @@ function LogDeprecatedUserAgent(string $requestType, ClientSupportLevel $clientS
                 }
             }
 
+            // If multiset is enabled, resolve the root game ID.
+            if (config('feature.enable_multiset')) {
+                $rootGameId = (new ResolveRootGameIdAction())->execute($gameHash, $game, $user);
```

Comment:
Note that `gameHash` will be null for multidisc games (see line 368). We do this so we don't spawn a new session when updating the rich presence string.

The client will send the hash of the currently loaded disc. Presumably, all discs would map to the same root game ID.